libtrellis/include/Bitstream.hpp

```diff
@@ -47,7 +47,7 @@ class Bitstream {
     static Bitstream serialise_chip(const Chip &chip, const map<string, string> options);
 
     // Deserialise a bitstream to a Chip
-    Chip deserialise_chip();
+    Chip deserialise_chip(const uint32_t idcode = 0);
```

Comment:
Please use `boost::optional<uint32_t>` as a zero idcode is as valid a value as any other.

Reply:
I'm not that familiar with boost, but I've given it a try. I wanted to ensure it was actually optional in the argument to avoid having to fix everywhere using this function.